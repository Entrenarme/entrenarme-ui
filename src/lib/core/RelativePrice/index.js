// @flow
import * as React from 'react';
import styled from 'styled-components';
import times from 'lodash/times';

import colors from '../../helpers/colors';
import { regularFont } from '../../helpers/fonts';

type Props = {
  currency?: string,
  price?: number,
};

const PriceItem = styled.span`
  color: ${props => (props.mark ? colors.text : colors.gray.default)};
  font-family: ${regularFont};
  font-size: 0.875rem;

  @media only screen and (max-width: 767px) {
    font-size: 0.75rem;
  }
`;

const RelativePrice = ({ currency, price }: Props) => {
  const renderPrice = (price, currency) => {
    const iterableArray = times(4, String);

    return iterableArray.map((item, index) => (
      <PriceItem key={item} mark={price > index}>
        {currency}
      </PriceItem>
    ));
  };

  return <div>{renderPrice(price, currency)}</div>;
};

RelativePrice.defaultProps = {
  currency: 'â‚¬',
  price: 0,
};

export default RelativePrice;
