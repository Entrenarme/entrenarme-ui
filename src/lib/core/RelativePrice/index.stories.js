// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';

import RelativePrice from './index';

storiesOf('RelativePrice', module).add(
  'default',
  withInfo('Relative price no props')(() => <RelativePrice />),
);

storiesOf('RelativePrice', module).add(
  'price prop',
  withInfo('Relative price with price')(() => <RelativePrice price={2} />),
);

storiesOf('RelativePrice', module).add(
  'price and currency prop',
  withInfo('Relative price with price and currency')(() => (
    <RelativePrice price={3} currency="$" />
  )),
);
