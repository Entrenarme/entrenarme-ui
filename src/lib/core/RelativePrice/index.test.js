import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import RelativePrice from './index';

test('renders "RelativePrice" with no props', () => {
  const { container } = render(<RelativePrice />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "RelativePrice" with price of 2', () => {
  const { container } = render(<RelativePrice price={2} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "RelativePrice" with price of 2 and using dollar currency', () => {
  const { container } = render(<RelativePrice price={3} currency="$" />);

  expect(container.firstChild).toMatchSnapshot();
});
