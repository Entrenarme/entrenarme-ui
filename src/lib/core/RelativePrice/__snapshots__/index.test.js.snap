// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "RelativePrice" with no props 1`] = `
<div>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    €
  </span>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    €
  </span>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    €
  </span>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    €
  </span>
</div>
`;

exports[`renders "RelativePrice" with price of 2 1`] = `
<div>
  <span
    class="sc-bwzfXH iNqkAx"
  >
    €
  </span>
  <span
    class="sc-bwzfXH iNqkAx"
  >
    €
  </span>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    €
  </span>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    €
  </span>
</div>
`;

exports[`renders "RelativePrice" with price of 2 and using dollar currency 1`] = `
<div>
  <span
    class="sc-bwzfXH iNqkAx"
  >
    $
  </span>
  <span
    class="sc-bwzfXH iNqkAx"
  >
    $
  </span>
  <span
    class="sc-bwzfXH iNqkAx"
  >
    $
  </span>
  <span
    class="sc-bwzfXH cEQODJ"
  >
    $
  </span>
</div>
`;
